### ISSUE TITLE
[Generic.WhiteSpace.ScopeIndent] Multi-line ternary false positive

### ISSUE THREAD

**Issue by bytestream** [squizlabs/PHP_CodeSniffer#3143]
**Describe the bug**
`Generic.WhiteSpace.ScopeIndent` reports an error when a multi-line ternary is written.

**Code sample**
Below is a clear succinct example, I stumbled across it on a significantly longer ternary which if on a single line would be `>120` characters and so not readable.

```php
<?php

$var = ! empty('var')
    ? true
    : false;

```

**Custom ruleset**
```xml
<?xml version="1.0"?>
<ruleset name="My Custom Standard">
  <rule ref="Generic.WhiteSpace.ScopeIndent">
    <properties>
      <property name="exact" value="true" />
    </properties>
  </rule>
</ruleset>
```

**To reproduce**
1. Create a file called `test.php` with the code sample above...
2. Run `phpcs test.php --standard=phpcs.xml`
3. See error message displayed
```
FILE: ~/test.php
-----------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 2 LINES
-----------------------------------------------------------------------
 4 | ERROR | [x] Line indented incorrectly; expected 0 spaces, found 4
 5 | ERROR | [x] Line indented incorrectly; expected 0 spaces, found 4
-----------------------------------------------------------------------
PHPCBF CAN FIX THE 2 MARKED SNIFF VIOLATIONS AUTOMATICALLY
-----------------------------------------------------------------------

Time: 25ms; Memory: 6MB

```

**Expected behavior**
No error is expected.

**Versions (please complete the following information):**
 - OS: Debian 9
 - PHP: 7.3.21
 - PHPCS: 3.5.6
 - Standard: Generic

**Comment by gsherwood** [comment]
You've enabled the `exact` property, which means that the sniff will enforce no indent for statements like this as a ternary is not a block opener.

I strongly recommend disabling the `exact` option because it removes your ability to use additional whitespace to make code easier to read.

**Comment by bytestream** [comment]
Ah OK.. I enabled the `exact`  property because the sniff doesn't error on the below without it:
```php
class Foo
{
    public function provideGetAllEndpoints(): iterable
    {
        foreach ($this->getGetAllEndpoints() as $endpoint => $apiCall) {
                yield [$endpoint, $apiCall];
        }
    }
}
```

**Comment by gsherwood** [comment]
Yes, that is expected. You can add more whitespace to lines but you can't have less than expected. There is no way to enforce that yield is indented exactly 1 stop from foreach without also asking the sniff to enforce a lot more exact checking.

**Comment by bytestream** [comment]
Gotcha. I'll just ignore it then 🤷‍♂️ 

Thanks for the quick replies!
